### PR TITLE
refactor(components): Update menu and popover for retheme [JOB-84566]

### PR DESCRIPTION
**Pull request by chris-at-jobber**
## Motivations

To enable the menu and popover components to handle alternate styling base on a retheme classname.

### Changed

- using gap instead of margin in both existing and retheme style to determine spacing between icon and label
- alternate styling available when `jobber-retheme` class is present on the body
  - spacing and typographic styles change on Menu
  - Popover has no arrow

## Testing

Add `jobber-retheme` class to the "story" iframe in the DOM

### Popover
- [ ] Looks correct _with_ retheme class
- [ ] Looks correct _without_ retheme class

#### Small screens
- [ ] Looks correct _with_ retheme class
- [ ] Looks correct _without_ retheme class

#### Medium+ screens
- [ ] Looks correct _with_ retheme class
- [ ] Looks correct _without_ retheme class

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Menu/Menu.css [hunk 1]
```diff
@@ -64,7 +64,7 @@
 
 .section {
   padding: var(--space-small) 0;
-  border-bottom: var(--border-base) dotted var(--color-border);
+  border-bottom: var(--border-base) solid var(--color-border);
```
Comment (chris-at-jobber) [inline packages/components/src/Menu/Menu.css:67]: This is a "just do it" change IMO

File: packages/components/src/Menu/Menu.css [hunk 2]
```diff
@@ -80,8 +80,23 @@
...
+:global(.jobber-retheme) .sectionHeader h6 {
+  /* can move into the Typography component post re-theme */
+  font-size: var(--typography--fontSize-base);
+  font-weight: 600;
+  text-transform: none;
+}
+
 .action {
   display: flex;
+  gap: var(--space-base);
```
Comment (chris-at-jobber) [inline packages/components/src/Menu/Menu.css:99]: removing the margin from Icon on the [old L100](https://github.com/GetJobber/atlantis/pull/1695/files#diff-487f2c6560890bbd027ea90576abb048193205478eecc1019326e553274973efL100) and converting to `gap` here

File: packages/components/src/Popover/Popover.css [hunk 3]
```diff
@@ -56,6 +59,10 @@
   clip-path: polygon(-4px -4px, 15.3px 0, 0 15.3px);
 }
 
+:global(.jobber-retheme) .arrow {
```
Comment (fagnanm) [inline packages/components/src/Popover/Popover.css:62]: @chris-at-jobber should popover keep the arrow like the SG-1 version? I'm not sure on it's usage and if the arrow is still necessary.

File: packages/components/src/Menu/Menu.css [hunk 4]
```diff
@@ -91,14 +106,25 @@
...
-.icon {
-  margin-right: var(--space-base);
+:global(.jobber-retheme) .action span {
+  /* can move into the Typography component post re-theme */
+  font-weight: 600;
+}
+
+:global(.jobber-retheme) .icon svg {
+  /* can move into the Icon component post re-theme */
+  fill: var(--color-interactive--subtle);
```
Comment (fagnanm) [inline packages/components/src/Menu/Menu.css:127]: This should be he darker/base blue, unless it changes to that in the theme?

File: packages/components/src/Menu/Menu.css [hunk 5]
```diff
@@ -91,14 +106,25 @@
...
 .action:hover,
-.action:focus {
+.action:focus-visible {
   background-color: var(--color-surface--hover);
   outline-color: var(--color-focus);
 }
 
-.icon {
-  margin-right: var(--space-base);
+:global(.jobber-retheme) .action span {
```
Comment (fagnanm) [inline packages/components/src/Menu/Menu.css:120]: This changes to text-secondary color

File: packages/components/src/Menu/Menu.css [hunk 6]
```diff
@@ -80,8 +80,23 @@
...
+  display: flex;
+  align-items: center;
+  min-height: var(--space-largest);
+  padding: var(--space-small) var(--space-base);
+}
+
+:global(.jobber-retheme) .sectionHeader h6 {
+  /* can move into the Typography component post re-theme */
+  font-size: var(--typography--fontSize-base);
+  font-weight: 600;
```
Comment (fagnanm) [inline packages/components/src/Menu/Menu.css:93]: Sorry forgot to update this in Figma, this remained un-bold or 400 weight

File: packages/components/src/Menu/Menu.css [hunk 7]
```diff
@@ -91,14 +106,25 @@
   align-items: center;
 }
 
+:global(.jobber-retheme) .action {
+  gap: var(--space-small);
+  padding: var(--space-small) calc(var(--space-small) + var(--space-smaller));
```
Comment (fagnanm) [inline packages/components/src/Menu/Menu.css:111]: To match SG-1 changes, use --space-small all around

File: packages/components/src/Menu/Menu.css [hunk 8]
```diff
@@ -80,8 +80,23 @@
   padding: var(--space-small);
 }
 
+:global(.jobber-retheme) .sectionHeader {
+  display: flex;
+  align-items: center;
+  min-height: var(--space-largest);
```
Comment (fagnanm) [inline packages/components/src/Menu/Menu.css:86]: To match SG-1 can drop the min height here, and instead use `padding: 12px 8px;`

**Review by fagnanm — CHANGES_REQUESTED** [review]
See comments inline

File: packages/components/src/Menu/Menu.css (hunk 6 above)
Comment (chris-at-jobber) [reply packages/components/src/Menu/Menu.css:93]: I don't think it did? 
image

Comment (chris-at-jobber) [reply packages/components/src/Menu/Menu.css:93]: Ohhh right this is the section header! On it

File: packages/components/src/Menu/Menu.css (hunk 4 above)
Comment (chris-at-jobber) [reply packages/components/src/Menu/Menu.css:127]: Yeah `interactive--subtle` maps to `blue` now

File: packages/components/src/Popover/Popover.css (hunk 3 above)
Comment (chris-at-jobber) [reply packages/components/src/Popover/Popover.css:62]: Hmm maybe

It is usually attached to an element, but so is Menu/Dropdown and we aren't using it there

I can see a case for leaving it

**Review by fagnanm — APPROVED** [review]
Looks good

**Pull request merged**